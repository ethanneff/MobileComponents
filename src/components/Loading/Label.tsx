import React, { useCallback, useEffect, useRef, useState } from "react";
import { StyleSheet, View, ViewStyle } from "react-native";
import Text from "../Text";

interface Props {
  title: string;
  center?: boolean;
  style?: ViewStyle;
}

const ellipsis = ["", ".", ".", "."];
const ellipsisDuration = 400;
const styles = StyleSheet.create({
  center: {
    justifyContent: "center"
  },
  row: {
    flexDirection: "row"
  }
});

export default function (props: Props) {
  const { title, center, style } = props;
  const ellipsisCountdown = useRef(0);
  const containerStyles = [
    styles.row,
    center ? styles.center : undefined,
    style
  ];
  const [ellipsisIndex, setEllipsisIndex] = useState(0);
  const animateTextNextIndex = (index: number) =>
    index >= ellipsis.length - 1 ? 0 : index + 1;
  const animateText = useCallback(() => {
    ellipsisCountdown.current = setTimeout(() => {
      setEllipsisIndex(index => animateTextNextIndex(index));
      animateText();
    }, ellipsisDuration);
  }, []);

  useEffect(() => {
    animateText();
    return () => clearTimeout(ellipsisCountdown.current);
  }, [ellipsisCountdown, animateText]);

  return (
    <View style={containerStyles}>
      <Text title={title} caption center />
      {ellipsis.map((dot, index) => (
        <Text
          key={index}
          title={dot}
          invisible={index > ellipsisIndex}
          caption
          center
        />
      ))}
    </View>
  );
}
