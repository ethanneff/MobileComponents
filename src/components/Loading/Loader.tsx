import React from "react";
import { Dimensions, StyleSheet, View, ViewStyle } from "react-native";
import DeviceInfo from "react-native-device-info";
import { Bar } from "react-native-progress";
import Theme from "../../commons/Theme";
import Icon from "../../components/Icon";

const isTablet = DeviceInfo.isTablet();
const iconSize = Theme.padding.p8;
const iconPadding = Theme.padding.p2;
const loaderCheckmarkSize = iconPadding * 2 + iconSize;

const styles = StyleSheet.create({
  container: {
    alignItems: "center",
    justifyContent: "center"
  },
  barContainer: {
    padding: Theme.padding.p3
  },
  iconContainer: {
    borderRadius: loaderCheckmarkSize,
    backgroundColor: Theme.color.success,
    padding: iconPadding,
    height: loaderCheckmarkSize,
    width: loaderCheckmarkSize,
    justifyContent: "center",
    alignItems: "center"
  }
});

interface Props {
  style?: ViewStyle | {};
  hidden?: boolean;
  animated?: boolean;
  finalComplete: boolean;
  indeterminate?: boolean;
  percent: number;
}

const WINDOW = Dimensions.get("window");

class Loader extends React.PureComponent<Props> {
  render() {
    const {
      hidden,
      finalComplete,
      percent,
      indeterminate,
      animated,
      style
    } = this.props;
    const width = WINDOW.height > WINDOW.width ? WINDOW.width : WINDOW.height;
    const containerStyle = style
      ? [styles.container, style]
      : [styles.container];
    const widthMultiplier = isTablet ? 0.5 : 0.6;
    return hidden ? null : (
      <View style={containerStyle}>
        {!finalComplete && (
          <View style={styles.barContainer}>
            <Bar
              animated={animated}
              testID="loaderProgressBar"
              height={isTablet ? 24 : 18}
              width={width * widthMultiplier}
              borderRadius={Theme.padding.p4}
              borderColor={Theme.color.neutral100}
              progress={percent}
              color={Theme.color.primary200}
              indeterminate={indeterminate}
            />
          </View>
        )}
        {finalComplete && (
          <View testID="loaderCheckmark" style={styles.iconContainer}>
            <Icon name={"check"} size={iconSize} color={Theme.color.neutral0} />
          </View>
        )}
      </View>
    );
  }
}

export default Loader;
