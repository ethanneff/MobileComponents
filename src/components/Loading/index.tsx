import React from "react";
import { StyleSheet, View, ViewStyle } from "react-native";
import DeviceInfo from "react-native-device-info";
import Theme from "../../commons/Theme";
import Button from "../Button";
import Text from "../Text";
import Label from "./Label";
import Loader from "./Loader";

interface Props {
  testID?: string;
  style?: ViewStyle | {};
  title?: string;
  label?: string;
  noLoading?: boolean;
  titleStyle?: ViewStyle | {};
  buttonTitle?: string;
  subtitle?: string;
  hidden?: boolean;
  isCompleted?: boolean;
  notAnimated?: boolean;
  isLoaderHidden?: boolean;
  indeterminate?: boolean;
  percent?: number;
  onButtonPress?(): void;
  onComplete?(): void;
}

interface State {
  finalComplete: boolean;
}

class Loading extends React.PureComponent<Props, State> {
  readonly state = {
    finalComplete: false
  };
  timeout?: number;
  readonly completionDelay = 1000;

  complete = () => {
    this.setState({ finalComplete: false });
  };

  componentDidUpdate() {
    const { isCompleted, percent = 0, onComplete } = this.props;
    const { finalComplete } = this.state;
    const complete = isCompleted || percent >= 1;

    if (finalComplete && !complete) {
      this.complete();
      return;
    }

    if (complete && onComplete && !finalComplete) {
      this.renderLastInterval();
    }
  }

  componentWillUnmount() {
    clearTimeout(this.timeout);
  }

  renderLastInterval() {
    clearTimeout(this.timeout);
    this.setState({ finalComplete: true });
    this.renderCheckMark();
  }

  renderCheckMark() {
    const { onComplete } = this.props;
    this.timeout = setTimeout(() => {
      clearTimeout(this.timeout);
      return onComplete && onComplete();
    }, this.completionDelay);
  }

  render() {
    const isTablet = DeviceInfo.isTablet();
    const styles = StyleSheet.create({
      container: {
        flex: 1,
        alignItems: "center",
        justifyContent: "center",
        paddingTop: Theme.sizing.vertical,
        paddingHorizontal: Theme.sizing.horizontal
      },
      title: {
        textAlign: "center",
        paddingBottom: isTablet ? Theme.padding.p5 : Theme.padding.p3
      },
      subtitle: {
        textAlign: "center",
        paddingTop: isTablet ? Theme.padding.p5 : Theme.padding.p3
      },
      button: {
        marginBottom: Theme.padding.p3
      }
    });
    const {
      style,
      title,
      titleStyle,
      buttonTitle = "",
      onButtonPress,
      subtitle,
      hidden,
      isLoaderHidden,
      indeterminate,
      notAnimated,
      label,
      noLoading,
      percent = 0
    } = this.props;
    const { finalComplete } = this.state;
    return hidden ? null : (
      <View style={[styles.container, style]}>
        {label && <Label title={label} />}
        <Text
          h3
          hidden={!title}
          style={[styles.title, titleStyle]}
          title={title}
        />
        <Button
          testID="loadingButton"
          hidden={!buttonTitle}
          half
          title={buttonTitle}
          onPress={onButtonPress}
          styleContainer={styles.button}
        />
        {!noLoading && (
          <Loader
            animated={!notAnimated}
            hidden={!!buttonTitle || isLoaderHidden}
            indeterminate={indeterminate}
            percent={percent}
            finalComplete={finalComplete}
          />
        )}
        <Text
          caption
          hidden={!subtitle}
          style={styles.subtitle}
          title={subtitle}
        />
      </View>
    );
  }
}

export default Loading;
