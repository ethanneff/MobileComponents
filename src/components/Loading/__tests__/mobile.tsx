import React from "react";
import renderer, { ReactTestRenderer } from "react-test-renderer";
import Loading from "..";
import { getTestProvider } from "../../../commons/Utils/TestMocks/get-test-provider";

describe("Loading Component", () => {
  let dom: ReactTestRenderer;

  it("renders correctly", () => {
    const { TestProvider } = getTestProvider();
    dom = renderer.create(
      <TestProvider>
        <Loading />
      </TestProvider>
    );
    expect(dom.toJSON()).toMatchSnapshot();
  });

  it("renders a title provided", () => {
    const { TestProvider } = getTestProvider();
    dom = renderer.create(
      <TestProvider>
        <Loading title="Some title" />
      </TestProvider>
    );
    expect(dom.toJSON()).toMatchSnapshot();
  });

  it("renders a subtitle provided", () => {
    const { TestProvider } = getTestProvider();
    dom = renderer.create(
      <TestProvider>
        <Loading subtitle="Some subtitle" />
      </TestProvider>
    );
    expect(dom.toJSON()).toMatchSnapshot();
  });

  it("renders nothing when hidden", () => {
    dom = renderer.create(<Loading buttonTitle="Some button title" hidden />);
    expect(dom.toJSON()).toMatchSnapshot();
  });

  it("renders nothing when isLoaderHidden", () => {
    const { TestProvider } = getTestProvider();
    dom = renderer.create(
      <TestProvider>
        <Loading buttonTitle="Some button title" isLoaderHidden />
      </TestProvider>
    );
    expect(dom.toJSON()).toMatchSnapshot();
  });

  it("renders a button if a button title is provided", () => {
    const { TestProvider } = getTestProvider();
    dom = renderer.create(
      <TestProvider>
        <Loading buttonTitle="Some button title" />
      </TestProvider>
    );
    expect(dom.toJSON()).toMatchSnapshot();
  });

  it("executes the button callback when the button is pressed", () => {
    const { TestProvider } = getTestProvider();
    let pressed = false;
    dom = renderer.create(
      <TestProvider>
        <Loading
          buttonTitle="Some button title"
          onButtonPress={() => {
            pressed = true;
          }}
        />
      </TestProvider>
    );
    const button = dom.root.findByProps({ testID: "loadingButton" });
    button.props.onPress();

    expect(pressed).toBe(true);
  });

  describe("when set as indeterminate", () => {
    it("renders as expected", () => {
      const { TestProvider } = getTestProvider();
      dom = renderer.create(
        <TestProvider>
          <Loading indeterminate={true} />
        </TestProvider>
      );
      expect(dom.toJSON()).toMatchSnapshot();
    });

    describe("once set as completed", () => {
      it("renders the checkmark once", () => {
        const { TestProvider } = getTestProvider();

        let isCompleted = false;

        dom = renderer.create(
          <TestProvider>
            <Loading
              indeterminate={true}
              isCompleted={isCompleted}
              onComplete={() => undefined}
            />
          </TestProvider>
        );

        isCompleted = true;
        dom.update(
          <TestProvider>
            <Loading
              indeterminate={true}
              isCompleted={isCompleted}
              onComplete={() => undefined}
            />
          </TestProvider>
        );

        expect(
          dom.root.findByProps({ testID: "loaderCheckmark" })
        ).toBeTruthy();
      });

      describe("once it times out", () => {
        jest.useFakeTimers();

        it("calls the onComplete callback", () => {
          const { TestProvider } = getTestProvider();
          let callbackExecuted = false;

          dom = renderer.create(
            <TestProvider>
              <Loading
                indeterminate={true}
                isCompleted={true}
                onComplete={() => {
                  callbackExecuted = true;
                }}
              />
            </TestProvider>
          );

          dom.update(
            <TestProvider>
              <Loading
                indeterminate={true}
                isCompleted={true}
                onComplete={() => {
                  callbackExecuted = true;
                }}
              />
            </TestProvider>
          );

          jest.runAllTimers();

          expect(callbackExecuted).toBe(true);
        });
      });

      describe("when set back to non complete", () => {
        it("renders the progress bar again", () => {
          const { TestProvider } = getTestProvider();

          let isCompleted = false;

          dom = renderer.create(
            <TestProvider>
              <Loading
                indeterminate={true}
                isCompleted={isCompleted}
                onComplete={() => undefined}
              />
            </TestProvider>
          );

          expect(
            dom.root.findByProps({ testID: "loaderProgressBar" })
          ).toBeTruthy();

          isCompleted = true;
          dom.update(
            <TestProvider>
              <Loading
                indeterminate={true}
                isCompleted={isCompleted}
                onComplete={() => undefined}
              />
            </TestProvider>
          );

          expect(
            dom.root.findAllByProps({ testID: "loaderProgressBar" })
          ).toHaveLength(0);

          isCompleted = false;
          dom.update(
            <TestProvider>
              <Loading
                indeterminate={true}
                isCompleted={isCompleted}
                onComplete={() => undefined}
              />
            </TestProvider>
          );

          expect(
            dom.root.findByProps({ testID: "loaderProgressBar" })
          ).toBeTruthy();
        });
      });
    });
  });

  describe("when set as determinate", () => {
    it("renders as expected", () => {
      const { TestProvider } = getTestProvider();
      dom = renderer.create(
        <TestProvider>
          <Loading indeterminate={false} />
        </TestProvider>
      );
      expect(dom.toJSON()).toMatchSnapshot();
    });

    describe("once progress is > 1", () => {
      it("renders the checkmark once", () => {
        const { TestProvider } = getTestProvider();

        let percent = 0;

        dom = renderer.create(
          <TestProvider>
            <Loading
              indeterminate={false}
              percent={percent}
              onComplete={() => undefined}
            />
          </TestProvider>
        );

        percent = 1;
        dom.update(
          <TestProvider>
            <Loading
              indeterminate={true}
              percent={percent}
              onComplete={() => undefined}
            />
          </TestProvider>
        );

        expect(
          dom.root.findByProps({ testID: "loaderCheckmark" })
        ).toBeTruthy();
      });

      describe("once it times out", () => {
        jest.useFakeTimers();

        it("calls the onComplete callback", () => {
          const { TestProvider } = getTestProvider();
          let callbackExecuted = false;

          dom = renderer.create(
            <TestProvider>
              <Loading
                indeterminate={true}
                percent={1}
                onComplete={() => {
                  callbackExecuted = true;
                }}
              />
            </TestProvider>
          );

          dom.update(
            <TestProvider>
              <Loading
                indeterminate={true}
                percent={1}
                onComplete={() => {
                  callbackExecuted = true;
                }}
              />
            </TestProvider>
          );

          jest.runAllTimers();

          expect(callbackExecuted).toBe(true);
        });
      });
    });
  });

  describe("when the loader is set as hidden", () => {
    it("renders as expected", () => {
      const { TestProvider } = getTestProvider();
      dom = renderer.create(
        <TestProvider>
          <Loading isLoaderHidden={true} />
        </TestProvider>
      );
      expect(dom.toJSON()).toMatchSnapshot();
      expect(
        dom.root.findAllByProps({ testID: "loaderCheckmark" }).length
      ).toEqual(0);
    });
  });
});
