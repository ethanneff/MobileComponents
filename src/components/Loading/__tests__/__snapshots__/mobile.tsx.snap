// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loading Component renders a button if a button title is provided 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
        "paddingHorizontal": 30,
        "paddingTop": 5,
      },
      undefined,
    ]
  }
>
  <TouchableOpacity
    accessibilityRole="button"
    activeOpacity={0.2}
    style={
      Array [
        Object {
          "borderRadius": 6,
          "justifyContent": "center",
          "padding": 12,
        },
        undefined,
        Object {
          "backgroundColor": "hsl(179, 58%, 67%)",
          "borderColor": "hsl(179, 58%, 67%)",
        },
        undefined,
        undefined,
        Object {
          "alignSelf": "center",
          "width": "80%",
        },
        undefined,
        undefined,
        undefined,
        undefined,
        undefined,
        Object {
          "marginBottom": 12,
        },
      ]
    }
    testID="loadingButton"
  >
    <View
      style={
        Array [
          Object {
            "flexDirection": "row",
            "justifyContent": "center",
          },
          undefined,
          undefined,
        ]
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        style={
          Array [
            Array [
              Object {
                "color": "hsl(200, 19%, 18%)",
                "fontFamily": "BrandonText-Bold",
                "fontSize": 15,
                "letterSpacing": 0.5,
                "lineHeight": 20,
              },
              Object {
                "textAlign": "center",
              },
              undefined,
              Object {
                "fontFamily": "BrandonText-Bold",
              },
              undefined,
              undefined,
              undefined,
              undefined,
            ],
            undefined,
          ]
        }
        textBreakStrategy="simple"
      >
        SOME BUTTON TITLE
      </Text>
    </View>
  </TouchableOpacity>
</View>
`;

exports[`Loading Component renders a subtitle provided 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
        "paddingHorizontal": 30,
        "paddingTop": 5,
      },
      undefined,
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "justifyContent": "center",
        },
      ]
    }
  >
    <View
      style={
        Object {
          "padding": 12,
        }
      }
    >
      <View
        testID="mockProgressBar"
      >
        <Text>
          Mock Progress Bar
        </Text>
        <Text
          testID="mockProgressBarProps"
        >
          {"animated":true,"testID":"loaderProgressBar","height":18,"width":450,"borderRadius":16,"borderColor":"hsl(198, 16%, 84%)","progress":0,"color":"hsl(179, 58%, 67%)"}
        </Text>
      </View>
    </View>
  </View>
  <Text
    accessible={true}
    allowFontScaling={true}
    style={
      Array [
        Array [
          Object {
            "color": "hsl(199, 18%, 33%)",
            "fontFamily": "BrandonText-Regular",
            "fontSize": 15,
            "letterSpacing": 0,
            "lineHeight": 20,
          },
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          Object {
            "paddingTop": 12,
            "textAlign": "center",
          },
        ],
        undefined,
      ]
    }
    textBreakStrategy="simple"
  >
    Some subtitle
  </Text>
</View>
`;

exports[`Loading Component renders a title provided 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
        "paddingHorizontal": 30,
        "paddingTop": 5,
      },
      undefined,
    ]
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    style={
      Array [
        Array [
          Object {
            "color": "hsl(200, 19%, 18%)",
            "fontFamily": "BrandonText-Medium",
            "fontSize": 22,
            "letterSpacing": 0,
            "lineHeight": 28,
          },
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          Array [
            Object {
              "paddingBottom": 12,
              "textAlign": "center",
            },
            undefined,
          ],
        ],
        undefined,
      ]
    }
    textBreakStrategy="simple"
  >
    Some title
  </Text>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "justifyContent": "center",
        },
      ]
    }
  >
    <View
      style={
        Object {
          "padding": 12,
        }
      }
    >
      <View
        testID="mockProgressBar"
      >
        <Text>
          Mock Progress Bar
        </Text>
        <Text
          testID="mockProgressBarProps"
        >
          {"animated":true,"testID":"loaderProgressBar","height":18,"width":450,"borderRadius":16,"borderColor":"hsl(198, 16%, 84%)","progress":0,"color":"hsl(179, 58%, 67%)"}
        </Text>
      </View>
    </View>
  </View>
</View>
`;

exports[`Loading Component renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
        "paddingHorizontal": 30,
        "paddingTop": 5,
      },
      undefined,
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "justifyContent": "center",
        },
      ]
    }
  >
    <View
      style={
        Object {
          "padding": 12,
        }
      }
    >
      <View
        testID="mockProgressBar"
      >
        <Text>
          Mock Progress Bar
        </Text>
        <Text
          testID="mockProgressBarProps"
        >
          {"animated":true,"testID":"loaderProgressBar","height":18,"width":450,"borderRadius":16,"borderColor":"hsl(198, 16%, 84%)","progress":0,"color":"hsl(179, 58%, 67%)"}
        </Text>
      </View>
    </View>
  </View>
</View>
`;

exports[`Loading Component renders nothing when hidden 1`] = `null`;

exports[`Loading Component renders nothing when isLoaderHidden 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
        "paddingHorizontal": 30,
        "paddingTop": 5,
      },
      undefined,
    ]
  }
>
  <TouchableOpacity
    accessibilityRole="button"
    activeOpacity={0.2}
    style={
      Array [
        Object {
          "borderRadius": 6,
          "justifyContent": "center",
          "padding": 12,
        },
        undefined,
        Object {
          "backgroundColor": "hsl(179, 58%, 67%)",
          "borderColor": "hsl(179, 58%, 67%)",
        },
        undefined,
        undefined,
        Object {
          "alignSelf": "center",
          "width": "80%",
        },
        undefined,
        undefined,
        undefined,
        undefined,
        undefined,
        Object {
          "marginBottom": 12,
        },
      ]
    }
    testID="loadingButton"
  >
    <View
      style={
        Array [
          Object {
            "flexDirection": "row",
            "justifyContent": "center",
          },
          undefined,
          undefined,
        ]
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        style={
          Array [
            Array [
              Object {
                "color": "hsl(200, 19%, 18%)",
                "fontFamily": "BrandonText-Bold",
                "fontSize": 15,
                "letterSpacing": 0.5,
                "lineHeight": 20,
              },
              Object {
                "textAlign": "center",
              },
              undefined,
              Object {
                "fontFamily": "BrandonText-Bold",
              },
              undefined,
              undefined,
              undefined,
              undefined,
            ],
            undefined,
          ]
        }
        textBreakStrategy="simple"
      >
        SOME BUTTON TITLE
      </Text>
    </View>
  </TouchableOpacity>
</View>
`;

exports[`Loading Component when set as determinate renders as expected 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
        "paddingHorizontal": 30,
        "paddingTop": 5,
      },
      undefined,
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "justifyContent": "center",
        },
      ]
    }
  >
    <View
      style={
        Object {
          "padding": 12,
        }
      }
    >
      <View
        testID="mockProgressBar"
      >
        <Text>
          Mock Progress Bar
        </Text>
        <Text
          testID="mockProgressBarProps"
        >
          {"animated":true,"testID":"loaderProgressBar","height":18,"width":450,"borderRadius":16,"borderColor":"hsl(198, 16%, 84%)","progress":0,"color":"hsl(179, 58%, 67%)","indeterminate":false}
        </Text>
      </View>
    </View>
  </View>
</View>
`;

exports[`Loading Component when set as indeterminate renders as expected 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
        "paddingHorizontal": 30,
        "paddingTop": 5,
      },
      undefined,
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "justifyContent": "center",
        },
      ]
    }
  >
    <View
      style={
        Object {
          "padding": 12,
        }
      }
    >
      <View
        testID="mockProgressBar"
      >
        <Text>
          Mock Progress Bar
        </Text>
        <Text
          testID="mockProgressBarProps"
        >
          {"animated":true,"testID":"loaderProgressBar","height":18,"width":450,"borderRadius":16,"borderColor":"hsl(198, 16%, 84%)","progress":0,"color":"hsl(179, 58%, 67%)","indeterminate":true}
        </Text>
      </View>
    </View>
  </View>
</View>
`;

exports[`Loading Component when the loader is set as hidden renders as expected 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
        "paddingHorizontal": 30,
        "paddingTop": 5,
      },
      undefined,
    ]
  }
/>
`;
